### PR TITLE
[doc] 0.12.2: prepare release notes

### PR DESCRIPTION
**Pull request by maflcko**
@laanwj Clearing the release notes was missing in the 0.12 branch.

Required to update the release notes.

### CODE REVIEW

**Comment by fanquake** [comment]
ACK ac129d2

File: doc/release-notes.md [hunk 1]
```diff
@@ -2,8 +2,7 @@ Bitcoin Core version 0.12.1 is now available from:
 
   <https://bitcoin.org/bin/bitcoin-core-0.12.1/>
 
-This is a new minor version release, including the BIP9, BIP68 and BIP112
-softfork, various bugfixes and updated translations.
+This is a new major version release, bringing new features and other improvements.
```
Comment (laanwj) [inline doc/release-notes.md]: I don't think this should be changed to major - 0.12.2 will yet again be a minor version release; there will never again be a major release from the 0.12 branch.

**Comment by laanwj** [comment]
Concept ACK

File: doc/release-notes.md (hunk 1 above)
Comment (MarcoFalke) [inline doc/release-notes.md]: Fixed.

**Pull request merged**